irods/irods_runtime_init.yml

```diff
@@ -34,6 +34,26 @@
         admin_password: "{{ _irods_clerver_password }}"
         zone: "{{ _irods_zone_name }}"
 
+    - name: Ensure that the starting collections get UUIDs
+      delegate_to: localhost
+      become: false
+      ansible.builtin.command:
+        cmd: "imeta ls -C {{ item }} | grep -q 'UUID' || imeta add -C {{ item }} UUID $(uuidgen -t)"
+        stdin: "{{ _irods_admin_password }}"
+      with_items:
+        - /{{ _irods_zone_name }}
+        - /{{ _irods_zone_name }}/home
+        - /{{ _irods_zone_name }}/home/{{ _irods_clerver_user }}
+        - /{{ _irods_zone_name }}/home/shared
+        - /{{ _irods_zone_name }}/trash
+        - /{{ _irods_zone_name }}/trash/home
+        - /{{ _irods_zone_name }}/trash/home/{{ _irods_clerver_user }}
+        - /{{ _irods_zone_name }}/trash/home/shared
+      # changed_when: "'UUID' not in cmd_result.stdout"
+      changed_when: false
```

Comment:
We need to detect when a change was made, i.e., when a UUID was assigned to a collection.

Reply:
Why would something like this lead to an idempotency test failure - if you uncomment the commented part.

Reply:
It won't break anything, but it enables change management/auditing. When a playbook is run and ansible indicates that a task changed something, the person running ansible can check to see if that task was supposed to changed something. A surprise change or no change can indicate a bad deployment that may need to be addressed manually.